pkg/distributor/distributor.go

```diff
@@ -605,6 +609,79 @@ func (d *Distributor) validateSeries(nowt time.Time, ts mimirpb.PreallocTimeseri
 	return nil
 }
 
+func (d *Distributor) pushWithMiddlewares() push.Func {
+	var middlewares []func(push.Func) push.Func
+
+	// The middlewares will be applied in the order of the slice "middlewares",
+	// requests will traverse them in the reverse order.
+	middlewares = append(middlewares, d.PrePushForwardingMiddleware)
+	middlewares = append(middlewares, d.PrePushHaDedupeMiddleware)
+
```

Comment:
We don't need to export `PrePushForwardingMiddleware` and `PrePushHaDedupeMiddleware` anymore.

Reply:
thx, fixed: a24c9a9